app/presenters/tree_node_builder.rb

```diff
@@ -376,6 +376,8 @@ def build_object_id
     else
       base_class = object.class.base_model.name           # i.e. Vm or MiqTemplate
       base_class = "Datacenter" if base_class == "EmsFolder" && object.kind_of?(Datacenter)
+      base_class = "ManageIQ::Providers::Foreman::ConfigurationManager" if base_class == "ExtManagementSystem" && object.kind_of?(ManageIQ::Providers::Foreman::ConfigurationManager)
```

Comment:
You don't need the `base_class == "ExtManagementSystem"` since you're checking with `kind_of?` and a subclass of ExtManagementSystem.